standard-cluster-services/src/main/java/org/terracotta/entity/IEntityMessenger.java

```diff
@@ -39,6 +39,18 @@
   void messageSelf(EntityMessage message) throws MessageCodecException;
 
   /**
+   * Defer a message based on another message, which will be sent in the future.
+   *
+   * @param tag String reason tag, mainly for debugging.
```

Comment:
Well, since this feature can leave to resource starvation, one can conceive of cases where you'd want to dump the list of outstanding explicit deferments, to see where you screwed up. This was a nice and clean way to denote "why" you did this. 

You can also foresee, if this becomes a problem, having the server monitor the outstanding explicit handles, and dump them to the log when they are a certain age, or even implementing a finalizer-esque  feature to release handles that have been GC'd. Or at least complain loudly about them. A small amount of context will go a long way.